src/helpers/counterBuilder.js

```diff
@@ -0,0 +1,221 @@
+      throw new Error('"digitMap" must be an object')
+    }
+    if (props.direction !== 'up' && props.direction !== 'down') {
+      throw new Error('"direction" must be either up or down')
+    }
+  },
+  /**
+   * Calculates initial state of the counter.
+   * @param {object} props
+   * @return {object} state
+   */
+  buildInitialState (counter) {
+    this.counter = counter
+    this.props = counter.props
+    this.state = {}
+    this.buildTimeProps()
+    this.buildPeriods()
+    this.buildMinAndMaxDigits()
+    this.buildCSSClasses()
+    this.buildInitialNumbers()
+    return this.state
+  },
+  /**
+   * Sets initial time parameters.
+   */
+  buildTimeProps () {
+    const startTime = new Date().getTime()
+    const from = (this.props.from === undefined) ? startTime : this.props.from
+    const timeDiff = (this.props.seconds === undefined)
+      ? (this.props.to - from)
+      : (this.props.seconds * 1000)
+    this.state = {
+      ...this.state,
+      startTime,
+      from,
+      timeDiff,
+      initialTimeDiff: timeDiff
+    }
+  },
+  /**
+   * Gets an array of periods to create segments for.
+   */
+  buildPeriods () {
+    const periods = PERIODS.slice(
+      PERIODS.indexOf(this.props.maxPeriod),
+      PERIODS.indexOf(this.props.minPeriod) + 1
+    )
+    this.state = {
+      ...this.state,
+      periods
+    }
+  },
+  /**
+   * Gets CSS classes for main counter.
+   */
+  buildCSSClasses () {
+    const type = this.props.easingFunction ? 'animated' : 'static'
+    var cssClasses = `rollex rollex-${type}`
+    if (this.props.frozen) cssClasses += ' rollex-frozen'
+    this.state = {
+      ...this.state,
+      cssClasses
+    }
+  },
+  /**
+   * Builds minDigits and maxDigits state properties.
+   */
+  buildMinAndMaxDigits () {
+    var [minDigits, maxDigits] = this.getMinAndMaxDigitsFromProps()
+    this.state = {
+      ...this.state,
+      ...this.normalizeMinAndMaxDigits(minDigits, maxDigits)
+    }
+  },
+  /**
+   * Gets minDigits and maxDigits from props specified by the user.
+   * @return {array}
+   */
+  getMinAndMaxDigitsFromProps () {
+    var minDigits = {}
+    var maxDigits = {}
+
+    if (typeof this.props.minDigits === 'number') {
+      for (let period of this.state.periods) minDigits[period] = this.props.minDigits
+    } else if (typeof this.props.minDigits === 'object') {
+      minDigits = this.props.minDigits
+    }
+    if (typeof this.props.maxDigits === 'number') {
+      for (let period of this.state.periods) maxDigits[period] = this.props.maxDigits
+    } else if (typeof this.props.maxDigits === 'object') {
+      maxDigits = this.props.maxDigits
+    }
+    return [minDigits, maxDigits]
+  },
+  /**
+   * Normalizes minDigits and maxDigits and sets default values where user did not specify them.
+   * @param {number|object} minDigits
+   * @param {number|object} maxDigits
+   * @return {object}
+   */
+  normalizeMinAndMaxDigits (minDigits, maxDigits) {
```

Comment:
var newMinDigits = {}, newMaxDigits = {}

const normalizeDigitsInPeriod = (min, max, mss) => {
  const minProvided = min !== undefined
  const maxProvided = max !== undefined
  const error (min, max) => {
    throw new Error(`conflict: minDigits (min) > maxDigits (max)`)
  }
  
  if(minProvided && maxProvided) { 
    min > max ? error(min, max) : return [min, max]
  } else if (minProvided && !maxProvided) {
    return [min, min > mss ? min : mss]
  } else if (!minProvided && maxProvided) {
    mss > max ? error(mss, max) : return [mss, max]
    // should it better be `return [mss > max ? max : mss, max]`?
  } else {
    [mss, mss]
  }
}

for ( let period of this.state.periods ) {
  const minSegmentSize = this.getMinSegmentSize(period)
  [newMinDigits[period], newMaxDigits[period]] = normalizeDigitsInPeriod(minDigits[period], maxDigits[period], minSegmentSize)
}

return { "minDigits": newMinDigits, "maxDigits": newMaxDigits }